doc/over_react_redux_documentation.md

```diff
@@ -0,0 +1,320 @@
+# OverReact Redux
+> A Dart wrapper for React Redux, providing targeted state updates.
+
+---
+* __[Purpose](#purpose)__
+    * [Benefits](#benefits)
+* __[Examples](#examples)__
+    * [Running the Examples](#running-the-examples)
+* __[Using it in your project](#using-it-in-your-project)__
+* __[ReduxProvider](#reduxprovider)__
+* __[Connect](#connect)__
+    * [`connect` Parameters](#connect-parameters)
+        * [mapStateToProps](#mapstatetoprops)
+        * [mapDispatchToProps](#mapdispatchtoprops)
+        * [mergeProps](#mergeprops)
+        * [areStatesEqual](#arestatesequal)
+        * [areOwnPropsEqual](#areownpropsequal)
+        * [areStatePropsEqual](#arestatepropsequal)
+        * [areMergedPropsEqual](#aremergedpropsequal)
+        * [context](#context)
+        * [pure](#pure)
+        * [forwardRef](#forwardref)
+* __[Using Multiple Stores](#using-multiple-stores)__
+* __[Using Redux DevTools](#using-redux-devtools)__
+
+## Purpose
+OverReact Redux is a Dart wrapper around [React Redux](https://react-redux.js.org), a UI binding library that allows for 
+better Redux integration with React. This integration allows for APIs that align more closely with React design patterns 
+while also providing more direct access to store data.
+
+### Benefits
+> For a full list of benefits, see <https://react-redux.js.org/introduction/why-use-react-redux>.
+ 
+While there are many benefits of using the library, a primary one is OverReact Redux allows for targeted updates 
+that only update components which receive data that has changed, rather than the whole component tree (as is the 
+behavior with React). By utilizing the `connect()` function in conjunction with `mapStateToProps()`, a component will 
+only update when a piece of information it uses is updated.
+
+## Examples
+Examples are available in the `web` directory. Each example illustrates a different variation or use case of OverReact
+ Redux. Additionally, the store files contain comments that call out specifics pertaining to that example and 
+ provides further explanation.
+ 
+### Running the Examples
+To run and experiment with the demo:
+1. `pub get`
+1. `pub run build_runner serve web`
+1. Navigate to `localhost:8080/over_react_redux/`
+
```

Comment:
Should we mention a way to see the isolated updates based on the "mapStateToProps" you check/turn on "Highlight updates when components render." from the react dev tools settings.
![Screen Shot 2019-10-31 at 10 57 53 AM](https://user-images.githubusercontent.com/41018730/67973753-576b5f00-fbce-11e9-9d10-0eaa989c69e9.png)

Reply:
💯 Good idea! I'll add that